lib/fields/class-wp-rest-meta-fields.php

```diff
@@ -222,7 +222,19 @@ protected function update_meta_value( $object, $name, $value ) {
 			);
 		}
 
-		if ( ! update_metadata( $this->get_meta_type(), $object, wp_slash( $name ), wp_slash( $value ) ) ) {
+		$meta_type  = $this->get_meta_type();
+		$meta_key   = wp_slash( $name );
+		$meta_value = wp_slash( $value );
+
+		// Do the exact same check for a duplicate value as in update_metadata() to avoid update_metadata() returning false.
+		$old_value = get_metadata( $meta_type, $object, $meta_key );
+		if ( 1 === count( $old_value ) ) {
```

Comment:
@joehoyle RE: the simpler conditional... technically yes (although a micro-optimization at best) - but the structure used by @tharsheblows here DOES indeed match the conditional structure already in `update_metadata()`